Allow symfony 6

akalineskou opened the pull request:
These were the changes I've made to allow symfony 6 to work.

The biggest one was `query->all` since `get` checks that the value is scalar and it was throwing an error since it was an array.
The other ones were deprecations for the return types.

akalineskou committed:
Update composer.json

akalineskou committed:
Update ConditionManager return values

akalineskou committed:
Update request query to use all for filter

denisvmedia commented on Query/ConditionManager.php:
```diff
@@ -54,27 +54,27 @@ public function offsetSet($offset, $value)
-    public function next() {
-        return next($this->conditions);
+    public function next(): void {
+        next($this->conditions);
```
What is the reason of this change? [Docs](https://www.php.net/manual/en/function.next.php) say:

> Returns the array value in the next place that's pointed to by the internal array pointer, or false if there are no more elements.

denisvmedia commented on composer.json:
```diff
@@ -12,9 +12,9 @@
   ],
   "require": {
     "php": ">=7.3.0",
```
You should then also increase php version to `>=8.0.0`. Because 7.3.* versions don't support typed properties, which you used in your PR. And php 7.4.* versions don't `mixed` type.

akalineskou replied:
Done

akalineskou committed:
Update composer.json

denisvmedia commented:
Note, as PHP 7.3 and 7.4 versions are now removed, we need to disable the tests for them.
We also need the tests for Symfony 6.0.

P.S. The code is fine, I will merge it, as soon as we have the tests green.

akalineskou committed:
Update github actions

akalineskou commented:
Done.

Updated symfony version 5 to 5.4 since the other ones are not supported anymore...
Also added php 8.1 for future releases (let me know if you dont want it tested)

akalineskou committed:
Fix tests

akalineskou commented:
Fixed the tests, also added try catch to the `get` in case of a `BadRequestException` to return the `InvalidArgumentException`

denisvmedia commented on Request/Request.php:
```diff
@@ -49,18 +50,42 @@ final class Request
-        if ($this->query !== null && !is_array($this->query)) {
+        try {
+            $this->pageNum = (int)$request->query->get('page', 1);
+        } catch (BadRequestException) {
```
Hm... Are you sure `$request->query->get` can return BadRequestException?

akalineskou replied:
It does throw one https://github.com/symfony/symfony/blob/6.1/src/Symfony/Component/HttpFoundation/InputBag.php#L37

The tests should also prove this since it returns the correct exception

denisvmedia replied:
Aha, ok. Thank you for this change.

denisvmedia reviewed (approved):
LGTM. Thank you for your efforts. Merging.

akalineskou commented:
Hey, when can we expect a new release/tag?

denisvmedia commented:
I will tag it tonight.
